src/utils/components/withMatchMedia.jsx

```diff
@@ -104,7 +104,9 @@ const withMatchMedia = WrappedComponent => {
 		 * @returns {React.element}
 		 */
 		render() {
-			return <WrappedComponent {...this.props} media={this.state.media} />;
+			const props = {...this.props};
+			delete props.initialMedia;
```

Comment:
The `delete` keyword should generally be avoided because it prevents some V8 compiler optimizations, so this should probably be written as

```jsx
render() {
  const { initialMedia, ...props } = this.props;  // eslint-disable-line no-unused
  return <WrappedComponent {...props} media={this.state.media} />;
}
```

Reply:
Ha I was going to do it that way until eslint complained

Reply:
yeah there's an old ticket in the ESLint repo about this behavior and how it shouldn't be warning in this case, but because object spread still isn't an official syntax spec (still Stage 3 (of 4)), the ESLint maintainers don't want to change the `no-unused` behavior yet :/